File: demo2-atp-cli/src/main/java/dev/kanashi/atp/cli/model/ExitCode.java
package dev.kanashi.atp.cli.model;

/**
 * 退出码是 agent 唯一可靠的分流依据，必须先定死再写命令层。
 *
 * <p>⚠️ 两条容易翻车的约定：
 * <ol>
 *   <li>幂等重放返回 {@link #OK}。重放在语义上是<b>成功</b> ——
 *       返回非 0 会让 agent 认为没成功而无限重试。</li>
 *   <li>{@link #VALIDATION_FAILED} 和 {@link #NEEDS_INPUT} 必须分开：
 *       前者 agent 自己能改，后者必须去问人。<b>下一步动作不同的，就不能合并成一个码。</b></li>
 * </ol>
 */
public enum ExitCode {

    OK(0),
    /** 版本对不上：用户确认之后内容被改过。重新 show → preview → 确认。 */
    VERSION_CONFLICT(10),
    /** 案例不存在，或草稿已被每月清理任务回收。 */
    NOT_FOUND(11),
    /** 规则校验不通过，agent 读 violations 自己改。 */
    VALIDATION_FAILED(12),
    /** 当前状态不允许该操作（如已 ACTIVE）。停下问用户。 */
    STATE_CONFLICT(13),
    /** 缺必填信息且机器补不出来。去问用户，不要猜。 */
    NEEDS_INPUT(14),
    /** DB 不通等基础设施故障。 */
    INFRA_ERROR(20);

    private final int code;

    ExitCode(int code) {
        this.code = code;
    }

    public int code() {
        return code;
    }
}

File: demo2-atp-cli/internal/model/exitcode.go
package model

// ExitCode 是 atp 对调用方（agent）的分派契约。
//
// ⚠️ 两条不能动的约定：
//
//  1. 幂等重放返回 OK —— 重放在语义上是成功；返回非 0 会让 agent 以为没成功而无限重试。
//  2. ValidationFailed 与 NeedsInput 必须分开 —— 一个是 agent 自己能改，一个必须去问人。
//     下一步动作不同的，就不能合并成一个码。
type ExitCode int

const (
	OK ExitCode = 0
	// VersionConflict 版本对不上：内容在用户确认之后被改过。重新 show → preview → 确认。
	VersionConflict ExitCode = 10
	// NotFound 案例不存在，或草稿已被每月清理任务回收。
	NotFound ExitCode = 11
	// ValidationFailed 值不合法，agent 读 violations 自己改。
	ValidationFailed ExitCode = 12
	// StateConflict 当前状态不允许该操作。停下，问用户。
	StateConflict ExitCode = 13
	// NeedsInput 缺必填信息且机器补不出来。去问用户，不要猜。
	NeedsInput ExitCode = 14
	// InfraError 配置缺失、数据库不通等。
	InfraError ExitCode = 20
)

func (c ExitCode) String() string {
	switch c {
	case OK:
		return "OK"
	case VersionConflict:
		return "VERSION_CONFLICT"
	case NotFound:
		return "NOT_FOUND"
	case ValidationFailed:
		return "VALIDATION_FAILED"
	case StateConflict:
		return "STATE_CONFLICT"
	case NeedsInput:
		return "NEEDS_INPUT"
	case InfraError:
		return "INFRA_ERROR"
	default:
		return "UNKNOWN"
	}
}
